k8srm-prototype/pkg/api/claim_types.go

```diff
@@ -36,15 +14,21 @@ type DeviceClass struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   DeviceClassSpec   `json:"spec,omitempty"`
-	Status DeviceClassStatus `json:"status,omitempty"`
+	Spec DeviceClassSpec `json:"spec,omitempty"`
 }
 
 // DeviceClassSpec provides the details of the DeviceClass.
 type DeviceClassSpec struct {
-	// DeviceType is a driver-independent classification of the device.
-	// This may be used instead of specifying the Driver explicitly, so that
-	// we do not aribtrarily limit claims to a particular vendor's devices.
+	// Driver specifies the driver that should handle this class of devices.
+	// When a DeviceClaim uses this class, only devices published by the
+	// specified driver will be considered.
+	// +required
+	Driver string `json:driver,omitempty`
+
+	// DeviceType is a driver-independent classification of the device.  In
```

Comment:
if Driver is required, is up to the scheduler to pick one or or other driver when user just wants to select by DeviceType?

Reply:
Yes. Scheduler evaluates (scores) possible solutions to the scheduling problem; this constrains the possible solutions but not as much as directly specifying the driver.